pennylane_qiskit/qiskit_device2.py

```diff
@@ -550,21 +550,37 @@ def _execute_runtime_service(self, circuits, session):
             else self.backend.configuration().backend_name
         )
 
-        options = {
+        circuit_runner_options = {
             "backend": backend_name,
             "log_level": self.options.environment.log_level,
             "job_tags": self.options.environment.job_tags,
             "max_execution_time": self.options.max_execution_time,
         }
 
+        # Does not support AerSimulator specific options e.g. choose a specific method
+        # refer to this https://qiskit.github.io/qiskit-aer/stubs/qiskit_aer.AerSimulator.html
+        # To support this would be confusing in terms of option setting (The options we currently support are runtime options)
+        # This here is just to capture and track shots information
+        aer_options = {
+            "shots": circuits[0].shots.total_shots or self.shots.total_shots,
```

Comment:
Instead of defining shots two places, could we add `shots = circuits[0].shots.total_shots or self.shots.total_shots` above in line 538, and then use it both in `program_inputs` and in the `aer` run function? Future updates are always harder when identical things are defined multiple places. 

Also instead of defining a full `options` dictionary if this is the only thing in it, should we just pass the `shots` kwarg in `self.backend.set_options(**aer_options)`?

Reply:
> Instead of defining shots two places, could we add `shots = circuits[0].shots.total_shots or self.shots.total_shots` above in line 538, and then use it both in `program_inputs` and in the `aer` run function? Future updates are always harder when identical things are defined multiple places.
> 
> Also instead of defining a full `options` dictionary if this is the only thing in it, should we just pass the `shots` kwarg in `self.backend.set_options(**aer_options)`?

Yeah I agree some refactoring is necessary here. I defined a full 'options' dictionary at first because I thought there would be more compatible options with the AerSimulator 😅 (seems like other than 'shots' nothing else though)